server/mqtt.go

```diff
@@ -959,7 +959,7 @@ func (s *Server) mqttCreateAccountSessionManager(acc *Account, quitCh chan struc
 		Retention: InterestPolicy,
 		Replicas:  as.replicas,
 	}
-	if _, err := jsa.createStream(cfg); isErrorOtherThan(err, ErrJetStreamStreamAlreadyUsed) {
+	if _, err := jsa.createStream(cfg); isErrorOtherThan(err, ApiErrors[JSStreamNameExistErr]) {
```

Comment:
Had this been in place, I would have checked for an error code here instead. The helper(s) createStream() and the like could be changed to return an error based on ApiError, but also the error code and that is what I would check here: (errors should be returned as last argument)
```
if _, errCode, err := jsa.createStream(cfg); errCode != "JS058" {
    return nil, fmt.Errorf("...", acc.GetName(), err)
}
```
(above is based on currently errCode be the ApiError.ErrCode which is a string as opposed to a number)

Reply:
It seems quite a verbose pattern to also return a code there - making go's already awful error handling more verbose and more returned values etc. 

Mainly I want to make as little as possible burdon on those writing functions, shifting the effort instead on their callers - most callers would never need to check a specific err code.

```
if _, err := foo(); !IsNatsErr(err, "JS058") { ... }
```

Wouldnt overly complicate the functions, allow us to use richer errors progressively in functions (what if `foo()` returned a ApiErr sometimes othertimes a go error? which is happening a bit in functions already).  Probably about the same level of effort but feels more go like.